miss_islington/status_change.py

```diff
@@ -100,16 +100,22 @@ async def check_ci_status_and_approval(
 
             title_match = TITLE_RE.match(normalized_pr_title)
             if title_match or is_automerge:
-                success = result["state"] == "success" and not any(
+                failure = any(
                     elem in [None, "failure", "timed_out"]
                     for elem in all_check_run_conclusions
                 )
+                success = result["state"] == "success" and not failure
```

Comment:
I would like to understand better what's going on here, and possibly capture this in a comment.
`result["state"]` seems to be the final outcome of all the checks combined.  I assume that some of the optional checks might fail or timeout without affecting this final result.  I'm also guessing that `all_check_run_conclusions` might include the optional checks, even though I'm not sure in which case the result is `None`.

If this assumptions are right, we have a few different possible outcomes:
* the status check is done, and it's a success
* the status check is done, it's a success, but some optional tests failed/timed out
* the status check is done, but it's a failure
* the status check is (not?) done, but it's a failure because of a timeout

We could therefore keep reporting the `result["state"]`, and then elaborate as described above.  We could also create two variables for `failures` and `timeouts` if we want a more accurate message.

What do you think?

Reply:
Best example is https://github.com/python/cpython/pull/94734/commits/40ccf1321b819ce383d23841868004c18d368eb0 from https://github.com/python/cpython/pull/94734 where I originally spotted this bug.

The `result['state']` comes from:
https://api.github.com/repos/python/cpython/commits/40ccf1321b819ce383d23841868004c18d368eb0/status

This refers to all checks from bedevere such as issue number and CLA signing.

`failure` is based on:
https://api.github.com/repos/python/cpython/commits/40ccf1321b819ce383d23841868004c18d368eb0/check-runs

These are all the Github Actions that get triggered. These do not include the checks that bedevere sets.

Thus, it is not so much about checks are required or optional, but rather about the source of the type of check.

Do we want to handle those differently?

Reply:
Is there any reason why the bedevere checks are listed separately from the other GitHub actions?  Are they not considered checks?

Reply:
Bedevere creates `statutes`, see https://docs.github.com/en/rest/commits/statuses

Since they are not triggered by a GitHub Action directly they need to be recovered from another endpoint. Honestly, the amount of different types of checks GitHub provides is quite long and their documentation is not optimal. Of the top of my head I can think of the terms actions, check suites, check runs, checks, statutes. There are probably more 😅